src/main/webapp/js/upload.js

```diff
@@ -55,16 +61,77 @@ async function fetchBlobstoreUrl() {
   return imageUploadUrl;
 }
 
+/**
+ * Converts the formatted price back to a number when the user
+ * selects the price input.
+ */
+function convertPricetoValue(event) {
+  const value = event.target.value;
+  event.target.value = value ? convertStringToNumber(value) : '';
+}
+
+/**
+ * Converts a string value into a number, removing all non-numeric characters.
+ */
+function convertStringToNumber(string) {
+  return Number(String(string).replace(/[^0-9.]+/g, ''));
+}
+
+/**
+ * Converts the number inputted by the user to a formatted string when
+ * the user unfocuses from the price input.
+ */
+function formatCurrency(event) {
+  const value = event.target.value;
+
+  if (value) {
+    event.target.value =
+        convertStringToNumber(value).toLocaleString(undefined, {
+          maximumFractionDigits: 2,
+          currency: 'USD',
+          style: 'currency',
+          currencyDisplay: 'symbol',
+        });
+  } else {
+    event.target.value = '';
+  }
+}
+
+/**
+ * Add the selected file name to the input label and check that the size of the
+ * uploaded file is within the limit.
+ */
+function displayFileName() {
+  const fileLabel = document.getElementById('receipt-filename-label');
+  const DEFAULT_FILE_LABEL = 'Choose file';
+
+  if (checkFileSize()) {
+    const fileInput = document.getElementById('receipt-image-input');
+    const fileName = fileInput.value.split('\\').pop();
+    fileLabel.innerText = fileName;
+  } else {
+    fileLabel.innerText = DEFAULT_FILE_LABEL;
+  }
+}
+
 /**
  * Displays an error message if the user selects a file larger than 5 MB.
+ * Returns true if the user selected a file with a smaller size.
```

Comment:
It would be nice to style the "returns true" line as described in the JS style guide:

@return {boolean} Whether the file size is less than 5 MB.

https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments